### PR TITLE
Remove reactive components

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Refactor the last uses of the Reactive Components, and disable their usage entirely.

#### Tests
Ran FPS Locally, will do cloud build.

#### Documentation
Added changelog

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 1]
```diff
@@ -1,54 +1,71 @@
+using System.Collections.Generic;
 using Improbable.Gdk.Core;
-using Improbable.Gdk.ReactiveComponents;
-using Unity.Collections;
+using Improbable.Worker.CInterop;
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: Why is this needed?

Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: For the `Authority.NotAuthoritative` on line 93

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ### Changed
 
 - Updated the `ConnectionController` and `ClientWorkerConnector` to make full use of the updated Player Lifecycle Feature Module.
+- Reactive Components are no longer used. They have been disabled through the use of the `DISABLE_REACTIVE_COMPONENTS` define symbol.
```
Comment (jamiebrynes7) [inline CHANGELOG.md]: >They have been disabled through the use of the `DISABLE_REACTIVE_COMPONENTS` define symbol. 

to

They have been disabled by adding the `DISABLE_REACTIVE_COMPONENTS` define symbol.

File: workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs [hunk 3]
```diff
@@ -1,59 +1,59 @@
...
-                            Origin = shotInfo.HitOrigin,
-                            AppliedLocation = shotInfo.HitLocation
-                        });
-                    commandSender.RequestsToSend.Add(modifyHealthRequest);
-                    commandSent = true;
+                    continue;
                 }
 
-                if (commandSent)
+                var shooterSpatialID = new EntityId(shotInfo.EntityId);
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs]: Why is this not an EntityId in the schema? 😢

File: workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs [hunk 4]
```diff
@@ -1,59 +1,59 @@
...
+                if (!workerSystem.TryGetEntity(shooterSpatialID, out var shooterEntity))
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs]: Does it matter which order `ValidateShot` and this check are done in? 

If not - we can put the less expensive one first

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 5]
```diff
@@ -1,54 +1,71 @@
...
+
+            initGroup = GetComponentGroup(ComponentType.ReadOnly<HealthRegenComponent.Component>(),
+                ComponentType.Subtractive<HealthRegenData>(),
+                ComponentType.ReadOnly<HealthComponent.ComponentAuthority>()
+            );
+            initGroup.SetFilter(HealthComponent.ComponentAuthority.Authoritative);
+
+            regenGroup = GetComponentGroup(
+                ComponentType.Create<HealthRegenComponent.Component>(),
+                ComponentType.Create<HealthRegenData>(), ComponentType.ReadOnly<HealthComponent.Component>(),
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: (Nit) Lets put one on each line otherwise your eyes skip over the second one when its the odd man out.

File: workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs [hunk 6]
```diff
@@ -9,54 +8,57 @@ namespace Improbable.Gdk.Health
...
+            workerSystem = World.GetExistingManager<WorkerSystem>();
+            commandSystem = World.GetExistingManager<CommandSystem>();
+            componentUpdateSystem = World.GetExistingManager<ComponentUpdateSystem>();
+        }
 
         protected override void OnUpdate()
         {
-            for (var i = 0; i < entitiesWithModifiedHealth.Length; i++)
+            var requests = commandSystem.GetRequests<HealthComponent.ModifyHealth.ReceivedRequest>();
+            var healthComponentData = GetComponentDataFromEntity<HealthComponent.Component>();
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs]: Can skip this if there are no requests to process

File: workers/unity/ProjectSettings/ProjectSettings.asset [hunk 7]
```diff
@@ -606,7 +606,7 @@ PlayerSettings:
   webGLLinkerTarget: 0
   webGLThreadsSupport: 0
   scriptingDefineSymbols:
-    1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP;IMPROBABLE_WORKERTYPE_UNITYGAMELOGIC;IMPROBABLE_WORKERTYPE_UNITYCLIENT;IMPROBABLE_WORKERTYPE_SIMULATEDPLAYERCOORDINATOR
+    1: UNITY_POST_PROCESSING_STACK_V2;UNITY_DISABLE_AUTOMATIC_SYSTEM_BOOTSTRAP;DISABLE_REACTIVE_COMPONENTS
```
Comment (jessicafalk) [inline workers/unity/ProjectSettings/ProjectSettings.asset]: why are they only disabled for PC

File: workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs (hunk 3 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs]: I know, but out of scope of this PR :'(

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs (hunk 5 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: Oops, forgot that here, will do

File: workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs (hunk 6 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs]: 👍

File: workers/unity/ProjectSettings/ProjectSettings.asset (hunk 7 above)
Comment (zeroZshadow) [reply workers/unity/ProjectSettings/ProjectSettings.asset]: Good question, will disable for all platforms. Might have to do this for playground as well

File: workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs (hunk 3 above)
Comment (jessicafalk) [reply workers/unity/Packages/com.improbable.gdk.guns/Systems/ServerShootingSystem.cs]: do we have a ticket for it?
